qutebrowser/browser/hints.py

```diff
@@ -720,6 +720,7 @@ def _handle_auto_follow(self, keystr="", filterstr="", visible=None):
                        if label.isVisible()}
 
         if len(visible) != 1:
+            self._apply_auto_follow_timeout()
```

Comment:
You'll definitely need to check `if len(visible) == 0` separately. Plus, as you already noticed, everything should go after the `if follow` check below.